adr/0003-websocket-next-client.adoc

```diff
@@ -0,0 +1,139 @@
+
+The execution model of each method depends on the method signature and the presence of the `@Blocking` annotation. If the method returns a `Uni`, `Multi`, `CompletionStage`, or `Publisher`, the method is considered as reactive and executed on the event loop. If the method returns a `void` or an object, the method is considered as blocking and executed on a worker thread. The `@Blocking` annotation can be used to force the method to be executed on a worker thread. The `@NonBlocking` annotation can be used to force the method to be executed on the event loop. Finally, blocking method can also be executed on a virtual thread by using the `@RunOnVirtualThread` annotation.
+
+The client endpoint feature allows to encapsulate the WebSocket client logic in a single class. Because it is a CDI bean, it can be easily injected in other beans, or use CDI events to communicate the received WebSocket messages with the rest of the application.
+
+=== WebSocket Connector
+
+The second concept introduced by the API is the WebSocket _connector_. The connector is used to configure and create new connections.
+While the client endpoint defines the methods to be called when an event occurs (message, connection opened), the connector is used to create the connection and configure it. The connector is a CDI bean and can be injected in other beans:
+
+Let's consider the following client endpoint:
+
+[source, java]
+----
+@WebSocketClient(path = "/endpoint/{name}")
+public static class ClientEndpoint {
+
+
+    @OnTextMessage
+    void onMessage(@PathParam String name, String message, WebSocketClientConnection connection) {
+        // ...
+    }
+
+    @OnClose
+    void close() {
+        // ...
+    }
+
+}
+----
+
+This endpoint is used as follows:
+
+[source, java]
+----
+// <1> Injection of the connector
+@Inject
+WebSocketConnector<ClientEndpoint> connector;
+
+// <2> Create the connection and configure the uri and path parameters (if any)
+WebSocketClientConnection connection = connector
+        .baseUri(uri)
+        .pathParam("name", "Roxanne")
+        .connectAndAwait();
+// <3> Use the connection to send messages if needed (the client endpoint can also retrieve the connection)
+connection.sendTextAndAwait("Hi!");
+----
+
+In this example, the connector is injected in a bean. The connector is used to create a new connection. The connection is configured with the base URI and path parameters. The connection is then established by calling the `connectAndAwait` method (asynchronous methods are also available). The connection can be used to send messages, close the connection, etc.
+
+The duality between the _client endpoint_ and the _connector_ separates the configuration and establishment of the websocket connection from the logic. The _client endpoint_ is used to define the logic of the WebSocket client, while the _connector_ is used to configure and create the connection.
```

Comment:
Should we also mention the `BasicWebSocketConnector` that can be used without a client endpoint class? 

https://github.com/quarkusio/quarkus/blob/main/extensions/websockets-next/deployment/src/test/java/io/quarkus/websockets/next/test/client/BasicConnectorTest.java

Reply:
Ah yes! Let me add this!

Reply:
Added as well as a note on the broadcast and the possibility to list the active connections.